16-legacy-code-workflow.md

```diff
@@ -0,0 +1,41 @@
+# Legacy Code Workflow
+
+We should establish a transition workflow for how we refactor and swap out legacy code components or encompassed-logic.
+
+**Discussion:** <https://github.com/DoSomething/rfcs/pull/16>
+
+## Problem
+
+When we are replacing an established component with a new component or replacing encompassed code logic in a function or class, the process may take multiple pull requests or steps to complete to avoid issues with the legacy code currently in use.
+
+As a developer proceeds through the process, they need a better way to transition replacing the existing code without causing downtime or confusing other developers as to which code approach should be used and which one is being deprecated.
+
+## Proposal
+
+To avoid downtime and remove confusion to developers the proposal is to begin the process by renaming the existing component/class/method/function and prefixing it with `Legacy*`. This would be the first pull request when tackling the task.
+
+For example, a component called `EmailSubscriptions` would be renamed to `LegacyEmailSubscriptions`, and all instances where it is used would be renamed as well in the first pull request for review.
+
+This would immediately highlight to other developers that the encompassed code is being deprecated and should potentially utilize the new version (ideally named the same, but without the `Legacy*` prefix).
+
+If the new encompassed code is renamed to something different, the legacy code with the `Legacy*` prefix could have a code comment indicating the new name of the code that is replacing it.
```

Comment:
Another use case that springs to mind (and is tangentially related to points below) is if we move the new component to a more appropriate location (from root to /utilities). This could be an opportunity to add another comment, or better yet, first, move the legacy component to its new location in the first place 🤔